jetbrains-core/src/software/aws/toolkits/jetbrains/services/federation/AwsConsoleUrlFactory.kt

```diff
@@ -0,0 +1,130 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.federation
+
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import org.apache.http.client.entity.UrlEncodedFormEntity
+import org.apache.http.client.methods.HttpGet
+import org.apache.http.impl.client.HttpClientBuilder
+import org.apache.http.message.BasicNameValuePair
+import software.amazon.awssdk.auth.credentials.AwsCredentials
+import software.amazon.awssdk.auth.credentials.AwsSessionCredentials
+import software.amazon.awssdk.core.client.config.SdkAdvancedClientOption
+import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.services.sts.StsClient
+import software.amazon.awssdk.services.sts.StsClientBuilder
+import software.aws.toolkits.core.region.AwsRegion
+import software.aws.toolkits.jetbrains.core.AwsClientManager
+import software.aws.toolkits.jetbrains.core.AwsClientManager.Companion.userAgent
+import software.aws.toolkits.jetbrains.core.AwsSdkClient
+import java.time.Duration
+
+class AwsConsoleUrlFactory(
+    private val httpClientBuilder: HttpClientBuilder = HttpClientBuilder.create(),
+    private val stsClientBuilder: StsClientBuilder = StsClient.builder()
+) {
+    fun consoleTld(region: AwsRegion) = when (region.partitionId) {
+        "aws-cn" -> {
```

Comment:
Is there any way we can get this automatically?

Reply:
for the console portion, i think we can just reuse the `dnsSuffix` parameter, but the signin endpoint depends on whether or not a region is opt-in and is not currently in the endpoints metadata, so maybe we should augment our endpoints generator to spit these out as well:

https://docs.aws.amazon.com/general/latest/gr/signin-service.html